pkg/core/blockchain.go

```diff
@@ -258,7 +258,7 @@ type transferData struct {
 // NewBlockchain returns a new blockchain object the will use the
 // given Store as its underlying storage. For it to work correctly you need
 // to spawn a goroutine for its Run method after this initialization.
-func NewBlockchain(s storage.Store, cfg config.Blockchain, log *zap.Logger) (*Blockchain, error) {
+func NewBlockchain(s storage.Store, cfg config.Blockchain, log *zap.Logger, nativeCfg ...native.ContractsConfig) (*Blockchain, error) {
```

Comment:
Why `ContractsConfig`? You can pass a constructor for a whole set of contracts (`NewContracts()` basically) and then do things you want with this structure. This would avoid `InitializeLinks()` and other hooks.